sparkly/catalog.py

```diff
@@ -38,25 +38,48 @@ def drop_table(self, table_name, checkfirst=True):
             table_name (str): A table name.
             checkfirst (bool): Only issue DROPs for tables that are presented in the database.
         """
+        db_name = get_db_name(table_name)
+        if checkfirst and not self.has_database(db_name):
+            return
+
         drop_statement = 'DROP TABLE IF EXISTS' if checkfirst else 'DROP TABLE'
-        return self._spark.sql('{} `{}`'.format(drop_statement, table_name))
+        return self._spark.sql(
+            '{} {}'.format(drop_statement, table_name)
+        )
 
-    def has_table(self, table_name, db_name=None):
+    def has_table(self, table_name):
         """Check if table is available in the metastore.
 
         Args:
             table_name (str): A table name.
-            db_name (str): A database name.
 
         Returns:
             bool
         """
+        db_name = get_db_name(table_name)
+        rel_table_name = get_table_name(table_name)
+
+        if not self.has_database(db_name):
+            return False
+
         for table in self._spark.catalog.listTables(db_name):
-            if table.name == table_name:
+            if table.name == rel_table_name:
                 return True
 
         return False
 
+    def has_database(self, db_name):
+        """Check if database exists in the metastore.
+
+        Args:
+            db_name (str): Database name.
+
+        Returns:
+            bool
+        """
+        return not db_name or db_name in [db.name
```

Comment:
Feel like someone spent too much time with clojure book. This line is hard to read, let's breakdown it into:
```
# Default database is always availabe
if db_name is None:
    return True

for db in self._spark.catalog.listDatabases():
    if db.name == db_name:
        return True

return False
```

If you still want to be a one-liner guru, let's make it more efficient:
```
db_name = 'default' if db_name is None else db_name
return any(db.name == db_name for db in self._spark.catalog.listDatabases())
```
Because in your case you create a list of all databases in memory.